packages/webdriverio/src/commands/element/isDisplayed.js

```diff
@@ -70,10 +70,11 @@ export default async function isDisplayed() {
      * no longer dictates it. In those instances, we pass the element through a script
      * that was provided by Brian Burg of safaridriver.
      */
-    return browser.isW3C && noW3CEndpoint.includes(browser.capabilities.browserName.toLowerCase()) ?
+    const result = browser.isW3C && !browser.isMobile && noW3CEndpoint.includes(browser.capabilities.browserName.toLowerCase()) ?
         await browser.execute(isElementDisplayedScript, {
             [ELEMENT_KEY]: this.elementId, // w3c compatible
             ELEMENT: this.elementId // jsonwp compatible
         }) :
         await this.isElementDisplayed(this.elementId)
+    return result === true
```

Comment:
@abjerstedt fix of Safari's `{ error: 'no such element', message: '', stacktrace: '' }`

Reply:
ignore this "fix".

Reply:
This won't fix the other commands though. We will want to update the response parser in webdriver.